active-rfcs/0000-style-script-behavior.md

```diff
@@ -0,0 +1,180 @@
+
+Note that this will continue to have issues with some use-cases. For example, if a component is used twice on the page, the dynamic value may change across different renders but would impact all components on the page. You can see an example of this here, where the last rendered component wins: https://stackblitz.com/edit/github-eww5sz?file=src%2Fcomponents%2FTour.astro&on=stackblitz
+
+
+## `<style global>`
+
+```astro
+<!-- INPUT: -->
+<style>h1 { color: red; }</style>
+
+<!-- OUTPUT: -->
+<style>h1 { color: red; }</style>
+```
+
+### Current Behavior
+
+- Same behavior described above, but not scoped to the component.
+
+## `<script>`
+
+```astro
+<!-- INPUT: -->
+<script>console.log('fired!');</style>
+
+<!-- OUTPUT: -->
+<script>console.log('fired!');</style>
+```
+
+### Current Behavior
+
+- Style content is escaped as JS but otherwise untouched by Astro. ex: TypeScript not supported.
+- ESM import will not be resolved or bundled, sent untouched/raw to the browser.
+- Will be output in the final component render(), useful for adding global JS.
+- May be duplicated if this component appears on the page multiple times.
+  - In the future, we could add some kind of of `is:unique` support to prevent duplicate scripts.
+
+## `<script hoist>`
+
+
+```astro
+<!-- INPUT: -->
+<script hoist>
+  import cowsay from 'cowsay';
+  console.log(cowsay('I am bundled with the rest of the page JS!'));
+</style>
+
+<!-- OUTPUT: -->
+<!-- JS is bundled with the rest of your page JavaScript. -->
+```
+
+
+### Current Behavior
+
+- Similar to `<script hoist>` today
+- Script content is processed, ex: TypeScript could be supported.
+- Script content is added to the module graph and bundled with the rest of the page JS.
+- ESM imports will be resolved using Vite's ESM import resolve logic.
+- No HTML output in the component template.
+- no HTML attributes supported (`type=`, `async`, `defer`, etc) because the element is virtual.
+
+### New RFC Behavior
+
+- `<script hoist>` contents must be static, therefore `define:vars` is not supported. Because the script is bundled ahead-of-time, dynamic values won't exist. This is currently easy to break in v0.21, so this RFC proposes removing the support entirely for `<script hoist>` (still available for `<script>` and `<style>`). 
+- Cannot be nested within a template expression or conditional. Bundled scripts must be scanned by the compiler. This is currently easy to break in v0.21 with something like `{alwaysFalse && <script hoist>...` so this RFC moves to remove support for this. This change shouldn't impact many users.
```

Comment:
How can we test for whether it will impact many users? Once usecase I would personally use for supporting LaTeX formatting would be conditionally adding in the CSS and JS for processing LaTeX markup only on pages with "hasMath" in the front-matter. I don't know how often that's a thing, but I definitely think it's a valid usecase!